container_linux.go

```diff
@@ -777,3 +815,37 @@ func (c *linuxContainer) currentState() (*State, error) {
 	}
 	return state, nil
 }
+
+// orderNamespacePaths sorts that namespace paths into a list of paths that we
+// can safely setns to.
+func orderNamespacePaths(namespaces map[configs.NamespaceType]string) ([]string, error) {
+	paths := []string{}
+	for _, nsType := range []configs.NamespaceType{
+		// TODO: enable configs.NEWUSER,
+		configs.NEWIPC,
+		configs.NEWUTS,
+		configs.NEWNET,
+		configs.NEWPID,
+		// mnt namespace must be the last one. After switching to mnt
+		// namespace, the old /proc will be inaccessible.
```

Comment:
hmm yes i think so. WDYT about opening all file descriptors first before `setns` so it's consistent. I think doing that helps with ordering too (i.e., order doesnt matter). Something like:
```
fds = []int{}
for ns in namespaces { fds = append(fds, open(ns) }
for fd in fds { setns(fd, 0) }
```